cpp/CMakeLists.txt

```diff
@@ -45,10 +45,17 @@ set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-endif(CMAKE_COMPILER_IS_GNUCXX)
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
+    option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" OFF)
```

Comment:
@kkraus14 @raydouglass @harrism doesn't this set the default value of the ABI flag to `OFF`?

I don't set the flag in the GPU test where it builds from source and I see this [output](https://gpuci.gpuopenanalytics.com/job/cudf-prb/1771/consoleFull#L320):
```
-- CUDF: Disabling the GLIBCXX11 ABI
-- ARROW: Disabling the GLIBCXX11 ABI
```

Reply:
I ask because the README change has it "defaults to ON"